### PR TITLE
Add pyxtal wrapper

### PR DESCRIPTION
**Pull request by pmrv**
[PyXtal](https://github.com/qzhu2017/PyXtal) is a similar structure generation tool as RandSPG, but (imo) more convenient and comprehensive.  I've made it a mandatory dependency, because I saw no reason not to, but I can wrap it in ImportAlarms as well.

### CODE REVIEW

**Comment by pmrv** [comment]
It seems I have to make it optional as mamba for some reason refuses to install another dependency for python compat reasons, even though that dependency (pyshtools) is available for up to python 3.8.

File: tests/atomistics/structure/test_pyxtal.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    def test_return_value(self):
+        """pyxtal should either return Atoms or StructureStorage, depending on arguments"""
+
+        self.assertIsInstance(pyxtal(1, species=['Fe'], num_ions=[1]), Atoms,
+                              "returned not an Atoms with scalar arguments")
+        self.assertIsInstance(pyxtal([1,2], species=['Fe'], num_ions=[1]), StructureStorage,
+                              "returned not a StructureStorage with multiple groups")
+        self.assertIsInstance(pyxtal(1, species=['Fe'], num_ions=[1], repeat=5),
+                              "returned not a StructureStorage with repeat given")
+        self.assertEqual(pyxtal(1, species=['Fe'], num_ions=[1], repeat=5), 5,
```
Comment (pmrv) [inline tests/atomistics/structure/test_pyxtal.py]:
```suggestion
        self.assertEqual(pyxtal(1, species=['Fe'], num_ions=[1], repeat=5).number_of_structures, 5,
```

**Comment by jan-janssen** [comment]
Is it possible to add `pyxtal` to https://github.com/pyiron/structuretoolkit ?

**Comment by jan-janssen** [comment]
The issue is that pyxtal requires `pyshtools ==4.10` which does not support Apple ARM and Python 3.11 while `pyshtools ==4.10.3` does. I raised an issue up stream to see if it is possible to add support for `pyshtools ==4.10.3` : https://github.com/qzhu2017/PyXtal/issues/222

**Comment by pmrv** [comment]
> Is it possible to add `pyxtal` to https://github.com/pyiron/structuretoolkit ?

Sure, I hadn't seen we had also moved the constructors over there.

**Comment by jan-janssen** [comment]
> Sure, I hadn't seen we had also moved the constructors over there.

As long as everything is ASE compatible, then it can go to `structuretoolkit`. The idea is that people who are already familiar with ASE can use the `structuretoolkit` without being forced to switch to pyiron completely. Once they like `structuretoolkit` then they might also be motivated to try pyiron.

File: .ci_support/environment.yml [hunk 2]
```diff
@@ -17,6 +17,7 @@ dependencies:
 - pyiron_base =0.6.1
 - pymatgen =2023.6.23
 - pyscal =2.10.18
+- pyxtal =0.5.7
```
Comment (jan-janssen) [inline .ci_support/environment.yml]:
```suggestion
- pyxtal =0.5.8
```

File: setup.py [hunk 3]
```diff
@@ -58,7 +58,8 @@
         'seekpath==2.1.0',
         'scikit-learn==1.2.2',
         'spglib==2.0.2',
-        'structuretoolkit==0.0.5'
+        'structuretoolkit==0.0.5',
+        'pyxtal==0.5.7'
```
Comment (jan-janssen) [inline setup.py]:
```suggestion
        'pyxtal==0.5.8'
```

**Comment by jan-janssen** [comment]
```
======================================================================
ERROR: test_return_value (atomistics.structure.test_pyxtal.TestPyxtal)
pyxtal should either return Atoms or StructureStorage, depending on arguments
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/pyiron_atomistics/pyiron_atomistics/tests/atomistics/structure/test_pyxtal.py", line 32, in test_return_value
    self.assertIsInstance(pyxtal(1, species=['Fe'], num_ions=[1], repeat=5),
  File "/usr/share/miniconda3/envs/my-env/lib/python3.9/unittest/case.py", line 1258, in assertIsInstance
    if not isinstance(obj, cls):
TypeError: isinstance() arg 2 must be a type or tuple of types

----------------------------------------------------------------------
```

File: tests/atomistics/structure/test_pyxtal.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+            pyxtal(194, ['Mg'], num_ions=[1], allow_exceptions=False)
+
+    def test_return_value(self):
+        """pyxtal should either return Atoms or StructureStorage, depending on arguments"""
+
+        self.assertIsInstance(pyxtal(1, species=['Fe'], num_ions=[1]), Atoms,
+                              "returned not an Atoms with scalar arguments")
+        self.assertIsInstance(pyxtal([1,2], species=['Fe'], num_ions=[1]), StructureStorage,
+                              "returned not a StructureStorage with multiple groups")
+        self.assertIsInstance(pyxtal(1, species=['Fe'], num_ions=[1], repeat=5),
```
Comment (jan-janssen) [inline tests/atomistics/structure/test_pyxtal.py]:
```suggestion
        self.assertIsInstance(pyxtal(1, species=['Fe'], num_ions=[1], repeat=5), StructureStorage,
```

**Comment by jan-janssen** [comment]
@pmrv With the new release of `pyxtal` all the tests finally pass, so I think we can move ahead with the integration in `pyiron_atomistics` or preferable `structuretoolkit`.

**Comment by pmrv** [comment]
Great, thanks for fixing the tests as well. I'll move the core functionality over to structuretoolkit next week then.

**Comment by pmrv** [comment]
Depends on https://github.com/pyiron/structuretoolkit/pull/24

File: .ci_support/environment.yml [hunk 5]
```diff
@@ -22,3 +22,4 @@ dependencies:
 - seekpath =2.1.0
 - spglib =2.0.2
 - structuretoolkit =0.0.6
+- pyxtal =0.5.9
```
Comment (jan-janssen) [inline .ci_support/environment.yml]: I am a bit reluctant to make `pyxtal` a required dependency as long as the environment doubles - as discussed in https://github.com/pyiron/structuretoolkit/issues/46 .

Comment (jan-janssen) [reply .ci_support/environment.yml]: This is also the reason why the upgrade from `structuretoolkit` is currently delayed https://github.com/pyiron/pyiron_atomistics/pull/1124

Comment (pmrv) [reply .ci_support/environment.yml]: Sure, this can wait until a new version of pyxtal is released.

**Comment by jan-janssen** [comment]
`structuretoolkit` 0.0.9 contains `pyxtal` 0.6.0

**Pull request merged**